src/python/pants/backend/python/check/pip_audit/rules.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from dataclasses import dataclass
+from hashlib import sha256
+from typing import Mapping
+
+from pants.backend.python.check.pip_audit.skip_field import SkipPipAuditField
+from pants.backend.python.check.pip_audit.subsystem import PipAudit
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import PythonRequirementResolveField, PythonRequirementsField
+from pants.backend.python.util_rules import pex_from_targets
+from pants.backend.python.util_rules.partition import ResolveName
+from pants.backend.python.util_rules.pex import PexRequest, PexResolveInfo, VenvPex, VenvPexProcess
+from pants.backend.python.util_rules.pex_from_targets import RequirementsPexRequest
+from pants.base.build_root import BuildRoot
+from pants.core.goals.check import CheckRequest, CheckResult, CheckResults
+from pants.engine.collection import Collection
+from pants.engine.fs import CreateDigest, Digest, FileContent
+from pants.engine.process import FallibleProcessResult, ProcessCacheScope
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import FieldSet, Target
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.ordered_set import FrozenOrderedSet, OrderedSet
+from pants.util.strutil import pluralize
+
+
+@dataclass(frozen=True)
+class PipAuditFieldSet(FieldSet):
+    required_fields = (
+        PythonRequirementsField,
+        PythonRequirementResolveField,
+    )
+
+    resolve: PythonRequirementResolveField
+    sources: PythonRequirementsField
```

Comment:
The name `sources` might be confusing because this refers to logical requirement strings, not source files. Maybe simply `requirements`?